File: Blind 75 LeetCode/Abinand P/Climbing Stairs/climbing stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if(n==0 || n==1){
            return 1;
        }
        int prev =1, current =1;
        for(int i=2;i<=n;i++){
            int temp =current;
            current = prev+current;
            prev = temp;
        }
        return current;
    }
};

File: Blind 75 LeetCode/Subhangi dhasmana/Climbing stairs/Climbing stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n==1 or n==2: return n
        dp = [-1]*(n+1)
        dp[1] = 1
        dp[2] = 2
        for i in range(3, n+1):
            dp[i] = dp[i-1]+dp[i-2]
        return dp[n]